components/BlogSection/index.tsx

```diff
@@ -1,21 +1,31 @@
 import PostsGrid from "@components/PostsGrid";
+import CategoryFilter from "@components/CategoryFilter";
 import { IPostProps } from "@components/Post";
 import styles from "./BlogSection.module.css";
 
 interface IAppProps {
   sectionTitle: string;
   posts: IPostProps[];
+  categories?: string[];
 }
 
-const BlogSection: React.FunctionComponent<IAppProps> = ({
+const BlogSection: React.FC<IAppProps> = ({
   sectionTitle,
   posts,
+  categories,
 }) => {
   return (
-    <div>
-      <h1>
-        <span className={styles.sectionTitle}>{sectionTitle}</span>
-      </h1>
+    <div className={styles.sectionContainer}>
+      <div className={styles.sectionHeader}>
+        <div>
+          <h1>
+            <span className={styles.sectionTitle}>{sectionTitle}</span>
+          </h1>
+        </div>
+        {Array.isArray(categories) && categories.length > 0 && (
```

Comment:
Okay! But...

What is the value of using `Array.isArray` having TypeScript? You declared categories as an array of strings already...

if you do something like this, you are going to get similar results because the prop categories can only be an array or undefined:

```suggestion
        {categories && categories.length > 0 && (
```

Reply:
Oh! Forgot about that one, my bad!